skorch/helper.py

```diff
@@ -245,7 +246,7 @@ def __getitem__(self, i):
             if i.dtype == np.bool:
                 i = np.flatnonzero(i)
 
-        return SliceDataset(self.dataset, idx=self.idx, indices=self.indices_[i])
+        return cls(self.dataset, idx=self.idx, indices=self.indices_[i])
```

Comment:
Same here regarding testing (but for `SliceDataset`).

Reply:
Good point, I added the unit tests.